src/pages/Home.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import axios from 'axios';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog, faMicrophone } from '@fortawesome/free-solid-svg-icons';
+import Header from '../components/Header';
+import StockCard from '../components/StockCard';
+import { saveStocksList, searchResults } from '../redux/store/stocks';
+import Apis from '../components/Apis';
+
+export default function Home() {
+  const stocksLists = useSelector((state) => state.stocks.stocksList);
+  const dispatch = useDispatch();
+  const savedData = JSON.parse(localStorage.getItem('stocksList'));
+  useEffect(() => {
+    if (savedData && savedData !== null) {
+      dispatch(saveStocksList(savedData));
+    } else {
+      axios.get(Apis.stocksList).then((res) => {
+        dispatch(saveStocksList(res.data));
+      }).catch((error) => {
+        console.error(error);
+        console.log('an error ocurred');
+      });
```

Comment:
- [ ] Why not remove this `console.log` which is mainly used for debugging and simply return the error response. You have moved your code to production and it is not a good idea to have a console log in the production environment. What do you think?

```suggestion
      }).catch((error) => error.response);
```